src/reducers/calculator.js

```diff
@@ -21,28 +22,36 @@ export const calculate = (displayValue, previusNumber, operator) => {
       return displayValue;
   }
 };
+
 const calculator = (state = initialState, action) => {
   switch (action.type) {
     case actionTypes.INPUT_NUMBER:
       return {
         ...state,
+        lastButton: action.number,
         displayValue:
           state.displayValue === "0"
             ? action.number.toString()
             : `${state.displayValue}${action.number}`
       };
     case actionTypes.INPUT_OPERATOR:
-      return {
+      let a = {
+        ...state,
         displayValue: "0",
-        operator: action.operator,
-        previusNumber: state.operator
+        operator: action.operator
+      };
+      let lastButton = a.lastButton;
+      if (lastButton !== a.operator) {
```

Comment:
As you directly accessing `lastButton` and do not need `lastButton` somewhere else, you could just directly use `a.lastButton` instead of first assign it to a new variable.